src/openrct2/interface/Console.cpp

```diff
@@ -298,8 +314,8 @@ void console_draw(rct_drawpixelinfo *dpi)
         sint32 caretX = x + gfx_get_string_width(lineBuffer);
         sint32 caretY = y + lineHeight;
 
-        uint8 caretColour = ColourMapA[BASE_COLOUR(backgroundColour)].lightest;
-        gfx_fill_rect(dpi, caretX, caretY, caretX + CONSOLE_CARET_WIDTH, caretY + 1, caretColour);
+        uint8 caretColour = ColourMapA[BASE_COLOUR(textColour)].lightest;
+        gfx_fill_rect(dpi, caretX, y + 9, caretX + CONSOLE_CARET_WIDTH, caretY, caretColour);
```

Comment:
The caret is still different compared to the text input window. Most notably, the caret is now quite thick when using e.g. Japanese — roughly twice as high as when using the sprite font. In the text input window, it is still only 1px high.

Reply:
That's because in all my `vector`-related excitement I screwed up the caret function! Fixed.